### DISCUSSION TITLE
Artisan not updating when using background playback

### DISCUSSION THREAD

**Discussion by andrewwdouglas** [artisan-roaster-scope/artisan#1412] [General]
I'm playing around with the playback events from an existing profile.  everything works fine except as soon the first event is triggered the graph stops updating.  I can force refresh the window (like minimizing and restoring) and the graph will be updated to that point but still won't live update.  the BT/ET, temps, etc... work fine and continue to update normally.  I'm using windows 10, Artisan 2.8.4 and a Kaleido Sniper M10 (connected via BT). The sampling rate is at the default value (1.50s) but it's not a sampling issue as it's still sampling at the set interval it's just not updating the graph.  My workaround is to toggle CTRL+F (fullscreen) which I can trigger quickly enough to update the graph, since the M10 has it's own control touch screen I can also just reference that screen too (but I would rather use the Artisan screen, and spit it out to whatever display I want.  If I'm not using the playback there's no issue with updating

**Comment by MAKOMO** [comment]
Maybe a regression. Could you please attach your Artisan Setttings here in a first step that we can duplicate your setup? For this, export the settings via menu Help >> Save Settings into an `artisan-settings.aset` file. Next rename that file to end in `.txt`. Finally, drag-and-drop this file to a comment box here in this thread on the GitHub Web page. A reply via Email will not work!

If possible, also send us your Artisan log file. For this click on the plus icon in the upper left corner of the main Artisan window while holding the ALT key. An email window should open with the log files attached. Please send us this email to `logfile@artisan.plus`.

Thanks for your support!

**Reply by andrewwdouglas** [reply]
below is the artisan-settings.txt file, I've also emailed the log files as requested.
[artisan-settings.txt](https://github.com/artisan-roaster-scope/artisan/files/14046054/artisan-settings.txt)

**Reply by MAKOMO** [reply]
Andrew, thanks for those settings and the log file. The issue seems to be a non-termination on redraw within the graph lib (or rather its numeric extension) according to your log. However, I could not reproduce this one on the Artisan v2.8.4 you are using and also nobody else did report this issue. There must be something special on your profile, maybe the background profile that provokes this bug. Could you also drag-drop-attach your background profile (and maybe one of your foreground profiles) as *.aset Artisan profile files here (again you need to rename them to end in *.txt?

The current Artisan v2.10 comes with an updated GUI lib which might have fixed that issue already and comes with many fixes and updates to the sample/redraw machine. The current development builds are based on a even newer version of that GUI lib. Maybe you could give our [current development build](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous) a go, which comes with many further fixes, and report back if it also suffers from that redraw issue. 

Thanks for your support!
